flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/TableLoader.java

```diff
@@ -34,12 +34,16 @@
  * the cluster (for example, to get splits), not just on the client side. So we need an Iceberg
  * table loader to get the {@link Table} object.
  */
-public interface TableLoader extends Closeable, Serializable {
+public interface TableLoader extends Closeable, Serializable, Cloneable {
 
   void open();
 
   Table loadTable();
 
+  /** By shallow cloning, clone a tableLoader. */
+  @SuppressWarnings("checkstyle:NoClone")
```

Comment:
instead of shallow cloning and suppress checkstyle warnings, can we implement proper clone. that would require `CatalogLoader` also implements `Cloneable` interface. From my quick look, it doesn't seem too bad.

cc @hililiwei @pvary for more comments.

Reply:
@xuzhiwen1255: If there's no strong reason against it, then I would go for the full clone too

Reply:
+1